peers/topicpool.go

```diff
@@ -23,8 +23,25 @@ const (
 // to get the maximum number of cached peers allowed.
 var maxCachedPeersMultiplier = 2
 
-// NewTopicPool returns instance of TopicPool
-func NewTopicPool(discovery Discovery, topic discv5.Topic, limits params.Limits, slowMode, fastMode time.Duration, cache *Cache) *TopicPool {
+// TopicPoolInterface the TopicPool interface.
+type TopicPoolInterface interface {
```

Comment:
This is a clear indicator that `TopicPool` needs to be refactored to meet the new requirements :)

Reply:
Btw. all these methods are required to be defined here?